### PR TITLE
JWT updates

### PR DESCRIPTION
**Pull request by alexkb0009**
Complement to https://github.com/dbmi-bgm/cgap-portal/pull/319

Ignore any edits in `es/...` folder, these files are auto-generated

### CODE REVIEW

File: src/components/browse/components/VirtualHrefController.js
Comment (alexkb0009) [inline src/components/browse/components/VirtualHrefController.js:199]: No longer needed here since is called after every AJAX request anyways via AJAXSettings.onLoad

File: src/components/util/ajax.js
Comment (alexkb0009) [inline src/components/util/ajax.js:170]: This probably worth removing and then maybe setting Cache headers if cache is explicitly true. Idk. To consider later. The current version of this doesn't work anyways (might get "/some/path/&ts=..." (not "?"). Is unrelated to this branch so will save for later.

File: src/components/navigation/components/LoginController.js
Comment (willronchetti) [inline src/components/navigation/components/LoginController.js:118]: if it's headed for timeout, you can bet on knowing about it after 30s.

File: src/components/navigation/components/LoginController.js
Comment (willronchetti) [inline src/components/navigation/components/LoginController.js:215]: While I do like inline comments like these, they aren't a replacement for docstrings IMO. The inline comments aren't always helpful if you don't understand how what it's doing fits into the overall scheme. This is more of a general comment as there are things you're doing to mitigate this ie: good function naming.

**Review by willronchetti — APPROVED** [review]
Approving with the caveat that I'm unlikely to catch JS specific errors. Generally the logic looks good.

File: src/components/navigation/components/LoginController.js
Comment (alexkb0009) [reply src/components/navigation/components/LoginController.js:91]: It's supposed to queue up the [code-split-out auth0 library JS file to be loaded in a tiny bit earlier than previously](https://webpack.js.org/guides/code-splitting/#prefetchingpreloading-modules). Not sure if it makes a practical difference since the auth0 library is queued to be loaded pretty early on anyway since is needed for the Login button to work.

Comment (alexkb0009) [reply src/components/navigation/components/LoginController.js:91]: [Another relevant article](https://medium.com/webpack/link-rel-prefetch-preload-in-webpack-51a52358f84c)

**Review by utku-ozturk — APPROVED** [review]
tested locally in Firefox/Safari/Opera and Chrome, looks and works great! Btw, Chrome's behavior is slightly different compared to other browsers when cookies are modified manually in inspect panel: The user is logged out of the portal but can not log in unless they clear cookies from settings. Not sure, but this is a feature of Chrome for extra security. This is not the case for Firefox/Safari and Opera.

Here is a screencast:
https://t.gyazo.com/teams/ugurteam/9b31b617de1c57ba4aeb73379b71ebfb.mp4

**Review by Bianca-Morris — APPROVED** [review]
Once you merge, I'll try to update my ajax method for formdata posting to be integrated with this (I might end up merging it into the current `ajax.promise` function... but for now I'll use what I currently have). I don't think any of this will conflict with it (merge or otherwise). You can have a closer look when you review my branches.

File: src/components/util/analytics.js
Comment (alexkb0009) [inline src/components/util/analytics.js:144]: We no longer have cookieStore (or the universal-cookie npm package, which has somewhat misleading documentation & functionality anyway); is simple enough to just do with raw JS. Maybe could set an expiry time as well, idk.

Comment (alexkb0009) [reply src/components/util/analytics.js:144]: Related - https://developer.mozilla.org/en-US/docs/Web/API/document/cookie

**Pull request merged**
